### PR TITLE
AdminSocket for alien_store

### PR DESCRIPTION
**Pull request by aclamk**
Introduced gluer that exposed classic admin socket commands through crimson admin socket.
This works only for alien_store.

It is based on #46618

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [x] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/crimson/os/alienstore/alien_admin_socket.cc [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+{
+ public:
+  CnAdminSocketHook(std::string prefix_str,
+                    std::string_view desc,
+                    std::string_view help,
+                    seastar::alien::instance& inst,
+                    unsigned shard,
+                    std::unique_ptr<crimson::os::ThreadPool>& tp,
+                    ::AdminSocketHook* hook)
+    :string_prefix(prefix_str)
```
Comment (athanatos) [inline src/crimson/os/alienstore/alien_admin_socket.cc:26]: Space after : and after ,

File: src/crimson/os/alienstore/alien_admin_socket.cc [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+static const std::string alien_prefix = "alien ";
+
+// Inheriting from string to preserve initialization order:
+// 1) prefix_str
+// 2) AdminSocketHook
+// We need to store prefix_str, since AdminSocketHook is directly storing string_view (no copy)
+class CnAdminSocketHook : public string_prefix, public crimson::admin::AdminSocketHook
+{
+ public:
```
Comment (athanatos) [inline src/crimson/os/alienstore/alien_admin_socket.cc:18]: public/private should not be indented

File: src/crimson/os/alienstore/alien_admin_socket.cc [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+  };
+};
+
+CnAdminSocket::CnAdminSocket(
+  ceph::common::CephContext* cct,
+  seastar::lw_shared_ptr<crimson::admin::AdminSocket> cn_asok,
+  std::unique_ptr<crimson::os::ThreadPool>& tp,
+  seastar::alien::instance& inst,
+  unsigned shard)
+:AdminSocket(cct)
```
Comment (athanatos) [inline src/crimson/os/alienstore/alien_admin_socket.cc:74]: Indent to level of unsigned and add spaces after : and ,

File: src/crimson/os/alienstore/alien_admin_socket.cc [hunk 4]
```diff
@@ -0,0 +1,131 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#include "alien_admin_socket.h"
+#include "crimson/admin/admin_socket.h"
+#include <seastar/core/alien.hh>
+
+using string_prefix = std::string;
+
+static const std::string alien_prefix = "alien ";
```
[the remark this answers is not in the document]
Comment (athanatos) [reply src/crimson/os/alienstore/alien_admin_socket.cc:10]: Not adding the additional alien prefix would also I think avoid needing to store a string in CnAdminSocketHook?

**Review by athanatos — CHANGES_REQUESTED** [review]
I think this approach is perfect -- it should pass through all of the BlueStore admin socket commands appropriately.

**Comment by aclamk** [comment]
jenkins test make check

**Comment by athanatos** [comment]
@aclamk I made some comments above, let me know if they make sense.  Also needs to be rebased.

**Pull request closed without merging**